qa/L0_http_v2/test.sh

```diff
@@ -0,0 +1,128 @@
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+set +e
+
+SIMPLE_V2_CLIENT=/workspace/install/bin/simple_v2_client
+
+RET=0
+
+# Build Server with HTTP V2 Support
+(cd /workspace/builddir && \
+    rm -fr trtis && \
+    cmake -DCMAKE_BUILD_TYPE=Release \
+        -DTRTIS_ENABLE_METRICS=OFF \
+        -DTRTIS_ENABLE_GCS=OFF \
+        -DTRTIS_ENABLE_S3=OFF \
+        -DTRTIS_ENABLE_CUSTOM=ON \
+        -DTRTIS_ENABLE_TENSORRT=OFF \
+        -DTRTIS_ENABLE_TENSORFLOW=ON \
+        -DTRTIS_ENABLE_CAFFE2=OFF \
+        -DTRTIS_ENABLE_ONNXRUNTIME=OFF \
+        -DTRTIS_ENABLE_ONNXRUNTIME_OPENVINO=OFF \
+        -DTRTIS_ENABLE_PYTORCH=OFF \
+        -DTRTIS_ENABLE_GPU=ON \
+        -DTRTIS_ENABLE_GRPC=OFF \
+        -DTRTIS_ENABLE_HTTP=ON \
+        -DTRTIS_ENABLE_HTTP_V2=ON \
+        ../build && \
+    make -j16 trtis && \
+    cp -r trtis/install/bin /opt/tensorrtserver/. && \
+    cp -r trtis/install/lib /opt/tensorrtserver/. && \
+    cp -r trtis/install/include /opt/tensorrtserver/include/trtserver)
+if [ $? -eq 0 ]; then
+    echo -e "\n***\n*** HTTP V2 Server Build Passed\n***"
+else
+    echo -e "\n***\n*** HTTP V2 Server Build Failed\n***"
+    RET=1
+fi
+
+# Install client dependencies
+(apt-get update && \
+    ln -fs /usr/share/zoneinfo/America/New_York /etc/localtime && \
+    export DEBIAN_FRONTEND=noninteractive
+    apt-get install -y tzdata && \
+    dpkg-reconfigure --frontend noninteractive tzdata && \
+    apt-get install -y --no-install-recommends \
+        libopencv-dev \
+        libopencv-core-dev \
+        pkg-config \
+        python3 \
+        python3-pip \
+        python3-dev && \
+    pip3 install --upgrade wheel setuptools grpcio-tools && \
+    ln -s /usr/bin/python3 /usr/bin/python)
+
+# Build Client with HTTP V2 Support
+(cd /workspace/build && \
+    cmake -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX:PATH=/workspace/install \
+        -DTRTIS_ENABLE_METRICS=OFF \
+        -DTRTIS_ENABLE_HTTP_V2=ON && \
+    make -j16 trtis-clients)
+if [ $? -eq 0 ]; then
+    echo -e "\n***\n*** HTTP V2 Client Build Passed\n***"
+else
+    echo -e "\n***\n*** HTTP V2 Client Build Failed\n***"
+    RET=1
+fi
+
+DATADIR=`pwd`/models
+SERVER=/opt/tensorrtserver/bin/trtserver
+SERVER_ARGS="--model-repository=$DATADIR"
+source ../common/util.sh
+
+run_server_nowait
```

Comment:
Just use run_server and then you don't need the sleep

Reply:
**run_server** calls **wait_for_server_ready** which repeatedly curls the old HTTP url ->  `localhost:8000/api/health/ready` till it gets a success code 200 and hence would get stuck here when using HTTP V2

Reply:
Can you put that comment in the script so we know why you are using it

Reply:
Sure